pkg/artifact/sbom/cyclonedx.go

```diff
@@ -0,0 +1,109 @@
+package sbom
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"os"
+	"path/filepath"
+
+	digest "github.com/opencontainers/go-digest"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/analyzer/config"
+	"github.com/aquasecurity/fanal/artifact"
+	"github.com/aquasecurity/fanal/cache"
+	"github.com/aquasecurity/fanal/handler"
+	"github.com/aquasecurity/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/sbom/cyclonedx"
+)
+
+var (
+	ArtifactCycloneDX types.ArtifactType = "cyclonedx"
+)
+
+type Artifact struct {
+	filePath       string
+	cache          cache.ArtifactCache
+	analyzer       analyzer.AnalyzerGroup
+	handlerManager handler.Manager
+
+	artifactOption      artifact.Option
+	configScannerOption config.ScannerOption
+}
+
+func NewArtifact(filePath string, c cache.ArtifactCache, opt artifact.Option) (artifact.Artifact, error) {
+	return Artifact{
+		filePath:       filepath.Clean(filePath),
+		cache:          c,
+		artifactOption: opt,
+	}, nil
+}
```

Comment:
We can determine which parser should be used in NewArtifact.

```suggestion
func NewArtifact(report, filePath string, c cache.ArtifactCache, opt artifact.Option) (artifact.Artifact, error) {
	var parser Parser
	switch report {
	case "cyclonedx":
		parser = cyclonedx.NewParser()
	}
	
	return Artifact{
		filePath:       filepath.Clean(filePath),
		cache:          c,
		artifactOption: opt,
	}, nil
}
```